src/compiler/checker.ts

```diff
@@ -11986,7 +11986,7 @@ namespace ts {
                             matchingCount = len;
                         }
                     }
-                    else if (!(overlap.flags & TypeFlags.Never) && 1 >= matchingCount) {
+                    else if (isUnitType(overlap) && 1 >= matchingCount) {

```

Comment:
`overlap.flags & TypeFlags.DisjointDomains` is probably better than straight `isUnitType` - this way we also capture `number` and `string` index signature overlap.

Reply:
That would weigh a numeric indexer equally with a common property, which isn't ideal.